core/win32/syscall.c

```diff
@@ -2961,7 +2964,13 @@ pre_system_call(dcontext_t *dcontext)
     reg_t *param_base = pre_system_call_param_base(mc);
     dr_where_am_i_t old_whereami = dcontext->whereami;
     dcontext->whereami = DR_WHERE_SYSCALL_HANDLER;
-    IF_X64(ASSERT_TRUNCATE(sysnum, int, mc->xax));
+    /* FIXME
+     * Commenting this ASSERT since x64 client injected in
+     * WOW64 binaries will be using mc->rax as comparaison
+     * thus making the assert fail, no matter if the
+     * syscall is correct or not.
+     */
+    // IF_X64(ASSERT_TRUNCATE(sysnum, int, mc->xax));
```

Comment:
The style guide disallows commented-out code (https://dynamorio.org/page_code_style.html#autotoc_md125).  It is also best to have an issue to reference with any FIXME or XXX comment.  If we don't understand this yet I think we want a ref to #49 and to disable based on just this situation.

```C
/* XXX i#49: mc->rax's top bits are non-zero in 32-bit mode for reasons we do not yet understand.
 * For now we disable the assert for mixed-mode.
 */
IF_X64(ASSERT(is_wow64_process(NT_CURRENT_PROCESS) || CHECK_TRUNCATE_TYPE_int(mc->xax)))    
```